src/schema/v2/me/collectArtistsMutation.ts

```diff
@@ -0,0 +1,91 @@
+import {
+  GraphQLList,
+  GraphQLNonNull,
+  GraphQLObjectType,
+  GraphQLString,
+  GraphQLUnionType,
+} from "graphql"
+import { mutationWithClientMutationId } from "graphql-relay"
+import { ResolverContext } from "types/graphql"
+import { UserInterest, userInterestType } from "../userInterests"
+
+interface Input {
+  artistIDs: string[]
+}
+
+const SuccessType = new GraphQLObjectType<any, ResolverContext>({
+  name: "collectArtistsSuccess",
+  fields: () => ({
+    userInterests: {
+      type: new GraphQLNonNull(new GraphQLList(userInterestType)),
+      resolve: (userInterests) => userInterests,
+    },
+  }),
+})
+
+const FailureType = new GraphQLObjectType<any, ResolverContext>({
+  name: "collectArtistsFailure",
+  fields: () => ({
+    mutationError: {
+      type: new GraphQLNonNull(GraphQLString),
+      resolve: (err) => err,
+    },
+  }),
+})
+
+const UserInterestsOrErrorType = new GraphQLUnionType({
+  name: "collectArtistsResponseOrError",
+  types: [SuccessType, FailureType],
+  resolveType: (data) =>
+    data._type === "GravityMutationError" ? FailureType : SuccessType,
+})
+
+export const collectArtistsMutation = mutationWithClientMutationId<
+  Input,
+  UserInterest[] | null,
+  ResolverContext
+>({
+  name: "CollectArtistsMutation",
+  description: "Collect Multiple Artists",
+  inputFields: {
+    artistIDs: {
+      type: new GraphQLNonNull(
+        new GraphQLList(new GraphQLNonNull(GraphQLString))
+      ),
+    },
+  },
+  outputFields: {
+    userInterestsOrError: {
+      type: UserInterestsOrErrorType,
+      description: "on success: returns the user interest",
+      resolve: (result) => result,
+    },
+  },
+  mutateAndGetPayload: async (args, { meCreateUserInterestLoader }) => {
+    if (!meCreateUserInterestLoader) {
+      throw new Error("You need to be signed in to perform this action")
+    }
+    const uniqueArtistIDs = Array.from(new Set(args.artistIDs))
```

Comment:
**suggestion:** We could use Lodash's [uniq](https://lodash.com/docs/4.17.15#uniq) here.

```suggestion
    const uniqueArtistIDs = uniq(args.artistIDs)
```